indexd/index/drivers/single_table_alchemy.py

```diff
@@ -0,0 +1,1548 @@
+                query.filter(Record.baseid == baseid)
+                .order_by(Record.created_date.asc())
+                .all()
+            )
+
+            # User requires update permissions for all versions of the record
+            all_resources = []
+            for rec in records:
+                all_resources += rec.authz
+            auth.authorize("update", list(all_resources))
+
+            ret = []
+            # Update fields for all versions
+            for record in records:
+                record.acl = set(acl) if acl else None
+                record.authz = set(authz) if authz else None
+
+                record.rev = str(uuid.uuid4())[:8]
+                ret.append(
+                    {"did": record.guid, "baseid": record.baseid, "rev": record.rev}
+                )
+            session.commit()
+            return ret
+
+    def get_latest_version(self, guid, has_version=None):
+        """
+        Get the lattest record version given did
+        """
+        with self.session as session:
+            query = session.query(Record)
+            query = query.filter(Record.guid == guid)
+
+            try:
+                record = query.one()
+                baseid = record.baseid
+            except NoResultFound:
+                baseid = guid
+            except MultipleResultsFound:
+                raise MultipleRecordsFound("multiple records found")
+
+            query = session.query(Record)
+            query = query.filter(Record.baseid == baseid).order_by(
+                Record.created_date.desc()
+            )
+
+            if has_version:
+                query = query.filter(Record.version.isnot(None))
+            record = query.first()
+            if not record:
+                raise NoRecordFound("no record found")
+
+            return record.to_document_dict()
+
+    def health_check(self):
+        """
+        Does a health check of the backend.
+        """
+        with self.session as session:
+            try:
+                query = session.execute("SELECT 1")  # pylint: disable=unused-variable
+            except Exception:
+                raise UnhealthyCheck()
+
+            return True
+
+    def __contains__(self, record):
+        """
+        Returns True if record is stored by backend.
+        Returns False otherwise.
+        """
+        with self.session as session:
+            query = session.query(Record)
+            query = query.filter(Record.guid == record)
+
+            return query.exists()
+
+    def __iter__(self):
+        """
+        Iterator over unique records stored by backend.
+        """
+        with self.session as session:
+            for i in session.query(Record):
+                yield i.did
+
+    def totalbytes(self):
+        """
+        Total number of bytes of data represented in the index.
+        """
+        with self.session as session:
+            result = session.execute(select([func.sum(Record.size)])).scalar()
```

Comment:
we could add a simple caching strategy here where we store some global and clear it if records get added. Maybe out of scope for this but worth considering. Same thing for num records